### PR TITLE
perf: persistent prompts and add chat record

### PR DESCRIPTION
**Pull request by jamebal**
1. Persistent cue word

2. Add chat log interface

### CODE REVIEW

**Comment by BobDu** [comment]
Sorry, before reviewing the logic changes, can you please merge the upstream main to the feature branch and then run lint?

In addition, please revert modifications to unrelated files such as gitignore and readme.

**Comment by jamebal** [comment]
1. already merge the upstream main branch.
2. already revert modifications to unrelated files such as gitignore and readme.

**Comment by BobDu** [comment]
Are you already run `pnpm lint:fix`?

**Comment by jamebal** [comment]
> Are you already run `pnpm lint:fix`?

yes

File: src/api/index.ts [hunk 1]
```diff
@@ -126,10 +126,10 @@ export function fetchRegister<T = any>(username: string, password: string) {
 }
 
 export function fetchUpdateUserInfo<T = any>(name: string, avatar: string, description: string) {
-  return post<T>({
-    url: '/user-info',
-    data: { name, avatar, description },
-  })
+	return post<T>({
```
Comment (BobDu) [inline src/api/index.ts:129]: So can you help check why eslint doesn't solve the indent format issue here?

Comment (jamebal) [reply src/api/index.ts]: I'm sorry. It's all fixed now.

**Comment by BobDu** [comment]
ping @jamebal 
Some comment in code review.

File: service/src/storage/config.ts [hunk 2]
```diff
@@ -117,8 +117,7 @@ export async function getOriginConfig() {
   }
 
   if (!isNotEmptyString(config.siteConfig.chatModels))
-    config.siteConfig.chatModels = 'gpt-3.5-turbo,gpt-4-turbo-preview,gpt-4-vision-preview'
-
+    config.siteConfig.chatModels = 'gpt-3.5-turbo,gpt-4,gpt-4-turbo,gpt-4o,gpt-4-turbo-preview,gpt-4-vision-preview,gemini-pro'
```
Comment (zhujunsan) [inline service/src/storage/config.ts:120]: `gemini-pro` shouldn't be here as default

**Review by BobDu — CHANGES_REQUESTED** [review]
Could you check for duplicate files that might not be used?

**Review by BobDu — APPROVED** [review]
LGTM

**Comment by zhujunsan** [comment]
better add index in mongodb:

```
db.chat.createIndex({ roomId: 1, dateTime: -1 });
```

**Pull request merged**